pkg/core/config/config.go

```diff
@@ -113,6 +121,91 @@ type Config struct {
+	IsEnabled bool `yaml:"is_enabled"`
+
+	// Services provides the OpenStack service-specific configuration.
+	Services OpenStackServices `yaml:"services"`
+
+	// Credentials specifies the OpenStack named credentials configuration,
+	// which is used by the various OpenStack services.
+	Credentials map[string]OpenStackCredentialsConfig `yaml:"credentials"`
+}
+
+type OpenStackServices struct {
+	// Compute provides the Compute service configuration.
+	Compute []OpenStackServiceConfig `yaml:"compute"`
+
+	// Network provides the Network service configuration.
+	Network []OpenStackServiceConfig `yaml:"network"`
+
+	// BlockStorage provides the Block Storage service configuration.
+	BlockStorage []OpenStackServiceConfig `yaml:"block_storage"`
+
+	// // ObjectStorage provides the Object Storage service configuration.
+	// ObjectStorage OpenStackServiceConfig `yaml:"object_storage"`
+}
+
+type OpenStackServiceConfig struct {
+	// UseCredentials specifies the named credentials to use.
+	UseCredentials string `yaml:"use_credential"`
+
+	// Domain specifies the domain to use when initializing the OpenStack client.
+	Domain string `yaml:"domain"`
+
+	// TODO: remove if project_id is enough. Will choose project ID, since it is unique
+	// across realms and can be used as a registry key.
+	// Project specifies the project to use when initializing the OpenStack client.
+	Project string `yaml:"project"`
+
+	// ProjectID specifies the project ID to use when initializing the OpenStack client.
+	ProjectID string `yaml:"project_id"`
+
+	// Region specifies the region to use when initializing the OpenStack client.
+	Region string `yaml:"region"`
+
+	// AuthEndpoint specifies the authentication endpoint to use when initializing an OpenStack client.
+	AuthEndpoint string `yaml:"auth_endpoint"`
+}
+
+type OpenStackCredentialsConfig struct {
+	// Authentication specifies the authentication method/strategy to use
+	// when creating OpenStack API clients.
+	// The currently supported authentication mechanisms are `user' for username/password
+	// and `app_credentials'.
+	Authentication string `yaml:"authentication"`
+
+	// User provides the settings to use for authentication when using username/password.
+	User OpenStackUserConfig `yaml:"user"`
+
+	// AppCredentials provides the settings to use for authentication when using application credentials.
+	AppCredentials OpenStackAppCredentialsConfig `yaml:"app_credentials"`
+}
+
+type OpenStackUserConfig struct {
+	// Username specifies the file path of the file containing the username to use.
+	UsernameFile string `yaml:"username_file"`
+
+	// Username specifies the file path of the file containing the password to use.
+	PasswordFile string `yaml:"password_file"`
+}
+
+type OpenStackAppCredentialsConfig struct {
+	// AppCredentialsIDFile specifies the file path containing the application credential ID to use when authenticating.
+	AppCredentialsIDFile string `yaml:"app_credentials_id_file"`
```

Comment:
Does the app id need to be stored in a file? If it is a public thing we can have it in the config as-is.